src/lando/api/legacy/workers/landing_worker.py

```diff
@@ -120,29 +133,15 @@ def process_merge_conflict(
         self,
         exception: PatchConflict,
         repo: Repo,
+        scm: AbstractSCM,
         revision_id: int,
     ) -> dict[str, Any]:
         """Extract and parse merge conflict data from exception into a usable format."""
         failed_paths, reject_paths = self.extract_error_data(str(exception))
 
         # Find last commits to touch each failed path.
         failed_path_changesets = [
-            (
-                path,
-                repo.hg.run_hg(
-                    [
-                        "log",
-                        "--cwd",
-                        repo.hg.path,
-                        "--template",
-                        "{node}",
-                        "-l",
-                        "1",
-                        path,
-                    ]
-                ),
-            )
-            for path in failed_paths
+            (path, scm.last_commit_for_path(repo.path, path)) for path in failed_paths
```

Comment:
Isn't `repo.path` going to be the same as `scm.path` (internally as `self.path`) here? If that's the case we could simplify the signature.